src/scenario/outdoors.cpp

```diff
@@ -199,3 +200,21 @@ bool cOutdoors::cWandering::isNull(){
 void cOutdoors::reattach(cScenario& scen) {
 	scenario = &scen;
 }
+
+bool cOutdoors::is_city_on_map(short tile_index)
+{
+	return city_locs[tile_index].spec >= 0 && city_locs[tile_index].x < AREA_MEDIUM;
+}
+
+bool cOutdoors::is_tile_on_edge(short tile_index)
+{
+	return city_locs[tile_index].x == minmax(0, AREA_MEDIUM - 1, city_locs[tile_index].x) &&
+		city_locs[tile_index].y == minmax(0, AREA_MEDIUM - 1, city_locs[tile_index].y);
+}
+
+unsigned short cOutdoors::get_terrain_index(short tile_index)
```

Comment:
This one... quite frankly it doesn't seem very useful, mainly because (once again) it takes an index into `city_locs` and then, besides that, it doesn't really do much of anything. Adding an `operator[]` in `vector2d` that takes `location` would probably be more useful (though I think I was purposely avoiding making `vector2d` depend on `location`...).

(That said, if kept you should be using the `ter_num_t` alias for the return type.)